pkg/controller/storagecluster/reconcile.go

```diff
@@ -138,17 +134,15 @@ func newStorageClassDeviceSets(dss []ocsv1alpha1.StorageDeviceSet) []rookCephv1.
 	return scds
 }
 
-func newStorageClassDeviceSet(ds *ocsv1alpha1.StorageDeviceSet) rookCephv1.StorageClassDeviceSet {
-	return rookCephv1.StorageClassDeviceSet{
+func newStorageClassDeviceSet(ds *ocsv1alpha1.StorageDeviceSet) rook.StorageClassDeviceSet {
+	return rook.StorageClassDeviceSet{
 		// XXX: We should possibly be generating and using a unique name here
 		Name:      ds.Name,
 		Count:     ds.Count,
 		Resources: ds.Resources,
 		Placement: ds.Placement,
 		// XXX: We should most likely be setting rook-ceph specific config options
 		// here, and not rely on user provided config. Just copying user provided config for now.
-		Config:               ds.Config,
-		VolumeClaimTemplates: ds.VolumeClaimTemplates,
+		VolumeClaimTemplates: []corev1.PersistentVolumeClaim{ds.DataPVCTemplate},
```

Comment:
We want to allow the admin to specify OSD configurations per device set. This will be critical for managing failure domain distribution, which will not be constant across all installations. Please leave `Config` in.

Reply:
I deleted the config line because I didn't want to write the function to convert from the struct to the map when I was testing builds. I should have added it back again, and added the conversion function, but I just forgot.